### ISSUE TITLE
Good first issues [meta-ticket]

### ISSUE THREAD

**Issue by Tomasz-W** [gravitystorm/openstreetmap-carto#3298]
### Regularly updated task list:
**single icons:**
- [x] amenity=boat_rental https://github.com/gravitystorm/openstreetmap-carto/issues/3263 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3300
- [x] amenity=bureau_de_change https://github.com/gravitystorm/openstreetmap-carto/issues/3185 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3301
- [x] shop=charity; second_hand https://github.com/gravitystorm/openstreetmap-carto/issues/1869 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3302
- [x] shop=interior_decoration https://github.com/gravitystorm/openstreetmap-carto/issues/1406 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3293
- [x] shop=video https://github.com/gravitystorm/openstreetmap-carto/issues/3116 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3313
- [x] leisure=bowling_alley https://github.com/gravitystorm/openstreetmap-carto/issues/3297 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3311
- [x] shop=paint https://github.com/gravitystorm/openstreetmap-carto/issues/3288 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3323
- [x] shop=massage https://github.com/gravitystorm/openstreetmap-carto/issues/3287 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3309
- [x] leisure=outdoor_seating https://github.com/gravitystorm/openstreetmap-carto/issues/3241 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3328
- [x] leisure=bird_hide https://github.com/gravitystorm/openstreetmap-carto/issues/3175 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3340
- [x] shop=bookmaker https://github.com/gravitystorm/openstreetmap-carto/issues/3186 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3345
- [x] shop=trade https://github.com/gravitystorm/openstreetmap-carto/issues/3291 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3361
- [x] man_made=storage_tank/ silo https://github.com/gravitystorm/openstreetmap-carto/issues/588 https://github.com/gravitystorm/openstreetmap-carto/issues/2909 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3384
- [x] orthodox cross ~~https://github.com/gravitystorm/openstreetmap-carto/issues/1186~~ -> closed
- [x] leisure=amusement_arcade https://github.com/gravitystorm/openstreetmap-carto/issues/3371 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3375
- [x] amenity=bicycle_repair_station https://github.com/gravitystorm/openstreetmap-carto/issues/3069 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3391
- [x] tourism=gallery https://github.com/gravitystorm/openstreetmap-carto/issues/883 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3401
- [x] amenity=vehicle_inspection https://github.com/gravitystorm/openstreetmap-carto/issues/3378 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3434
- [x] amenity=internet_cafe https://github.com/gravitystorm/openstreetmap-carto/issues/3451 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3463
- [x] man_made=telescope https://github.com/gravitystorm/openstreetmap-carto/issues/3389 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3473
- [x] amenity=public_bookcase https://github.com/gravitystorm/openstreetmap-carto/issues/1288 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3462
- [x] leisure=fishing https://github.com/gravitystorm/openstreetmap-carto/issues/3390 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3508
- [x] shop=fabric/ carpet/ sewing https://github.com/gravitystorm/openstreetmap-carto/issues/3289 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3555

**sets of icons:**
- [x] memorials https://github.com/gravitystorm/openstreetmap-carto/issues/3088 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3356
- [x] healthcare objects https://github.com/gravitystorm/openstreetmap-carto/issues/2938 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3498
- [x] crane https://github.com/gravitystorm/openstreetmap-carto/issues/3478 -> done! https://github.com/gravitystorm/openstreetmap-carto/pull/3501 -> cranes ~~https://github.com/gravitystorm/openstreetmap-carto/issues/3573~~ -> closed :(

**Comment by Tomasz-W** [comment]
_<place for further goals>_

**Comment by lakedistrictOSM** [comment]
@Tomasz-W Have you thought about writing an osm.org Diary Entry or /r/openstreetmap post to see if there's anyone who's able to do the coding and testing for these prepared icons? :)

**Comment by Martin-Lbg** [comment]
I might help in coding, but since I never coded anything for OSM, I would need some instructions. I'm able to program in general and I'm willing to learn OSM-methodolgy - but got stuck in the vast amount of different actions needed (at least it Looks that way to me). Where could I start?

**Comment by kocio-pl** [comment]
Great! 

Two links that works for most of the newcomers are:

https://wiki.openstreetmap.org/wiki/Standard_tile_layer
https://github.com/gravitystorm/openstreetmap-carto/blob/master/DOCKER.md

If you however will need some help, just ask here.

**Comment by Tomasz-W** [comment]
here you are: https://www.openstreetmap.org/user/Tomasz_W/diary/44420

**Comment by Adamant36** [comment]
@Tomasz-W Here's some more issues that need code and testing to add to your list.  

#1547
#2431
#2691
#3274
#3169
#3073
#1984
#1943
#1889
#1874
#1547
#3034
#2905
#2896
#2411
#1977
#1934
#1746
#1659
#585
#552

**Comment by Adamant36** [comment]
@Tomasz-W I dont disagree. I never said they needed to be added now or that the  simpler tasks cant be done first. So feel free to add them after the simpler ones are done. At least they are there in the mean time though if someone wants give them a try.

**Comment by kocio-pl** [comment]
For me it's just a @Tomasz-W list - he's doing some managing things at the moment, so it's up to him how he does like to do it. I'm happy to see people involved in different activities, because we need not only coders. This list is a social tool helping newcomers to start - it looks that they feel more welcome this way, so we just go with it.

That being said, I try to not stick to just one type of things. Sometimes I do some trivial fixes to just relax a bit within a project and have some fast reward, because that's encouraging, but we have also some big challenges that are important and I also treat them as priority, but at the moment nobody is working on them (see https://github.com/gravitystorm/openstreetmap-carto/projects).

**Comment by Tomasz-W** [comment]
I just want to inform some of previous contributors that we still need coders for further map developing. Please take a look at the first two posts in this issue (https://github.com/gravitystorm/openstreetmap-carto/issues/3298#issue-340348239 https://github.com/gravitystorm/openstreetmap-carto/issues/3298#issuecomment-405039682). Maybe there will be something that you will be interested in writing code for. We need YOU! :)

@skylerbunny @d3d9 @hikemaniac @vnbhattarai @MKuranowski @jnachtigall @systemed @boothym @davidfg4 @PissedCapslock @sarxos @mojodna @tmcw @derickr @DeeHants @AurimasF @micechal @CloCkWeRX @Rovastar @chrisfleming @mstock @wonderchook @mackerski @tyrasd @smsm1 @M1dgard @ubahnverleih @littlebtc @doktorpixel14 @jbelien @MattBlissett @Firefishy @woodpeck @kaldari @erictheise @AndrewHain @jdhoek @tpikonen @altoing @Zverik @HolgerJeromin @micahcochran @andrzej-r @Penegal @kreed @vholten @StyXman @floscher @mrwojo @danstowell @Ircama  @dkniffin  @ian29

**Comment by systemed** [comment]
@Tomasz-W please don't spam people's mentions like that, seriously not cool

**Comment by Tomasz-W** [comment]
**list of issues closed without test renderings, when there was still some people interested in rendering them:**

- [ ] tourism=gallery https://github.com/gravitystorm/openstreetmap-carto/issues/883
- [ ] amenity=coworking_space https://github.com/gravitystorm/openstreetmap-carto/issues/1628
- [ ] amenity=studio https://github.com/gravitystorm/openstreetmap-carto/issues/2518
- [ ] chess tables https://github.com/gravitystorm/openstreetmap-carto/issues/3039
- [ ] benches with backrest https://github.com/gravitystorm/openstreetmap-carto/issues/3187
- [ ] leisure=bleachers https://github.com/gravitystorm/openstreetmap-carto/issues/3140

**Comment by Tomasz-W** [comment]
@Adamant36 Actually it's a list of issues where we need a test renderings at least, to decide what's next with it (rate readability of the icon on map etc.). Doing the same thing in PRs is also ok and it makes the process faster, so I prefer PRs.

**Comment by Adamant36** [comment]
The name of the issue is "Code Needed to Rendered Prepared Icons" and it seems like the list isn't at that point anymore. Since most of the icons that are "prepared" have already merged and most things either need test rendering now or more debate. So can you at least update the title to reflect that the issue about that anymore? It might be a minor thing, but it would still be nice. 

I totally agree with you on PRs. They have the advantage of going faster in a lot of cases. A lot of the unsolved issues here that are stuck in debate could probably be dealt with if someone just did PRs for them. Even if they aren't perfect solutions.

**Comment by Adamant36** [comment]
@Tomasz-W, what do you think the priorties are at this point and what would you like to see me work on next? Im feeling pretty good about the stuff we have gotten done so far but im not sure whats worth focusing on now.

**Comment by Tomasz-W** [comment]
My personal favourities at the moment are #3326 (I hope with https://github.com/gravitystorm/openstreetmap-carto/issues/3187), https://github.com/gravitystorm/openstreetmap-carto/issues/3143 + https://github.com/gravitystorm/openstreetmap-carto/issues/2098,  and https://github.com/gravitystorm/openstreetmap-carto/issues/2086 . I'm waiting for most of icons from first post, but I'm not going to copy-paste here whole list, I would be happy if any of them will be pushed forward.

**Comment by Adamant36** [comment]
Alright. The backrest one clearly isnt going to happen, but I think the other ones are doable.

**Comment by Tomasz-W** [comment]
@Adamant36 It would be also good to finish abandoned https://github.com/gravitystorm/openstreetmap-carto/pull/3135 and https://github.com/gravitystorm/openstreetmap-carto/pull/3126. It looks like @james2432 left it.

**Comment by Adamant36** [comment]
Good call. Those have bugging me for a while now. I'll work on them next. Id love to see more of the stagnate PRs get dealt with.

**Comment by Adamant36** [comment]
You think I should wait on craft until the color and #3395 is worked out? It doesnt seem like the color has been decided yet. Maybe I could do some test renderings of different colors or something in the mean time if nothing else.

**Comment by Tomasz-W** [comment]
@Adamant36 I think that better way is to firstly see a test rendering with different colours, and then discuss and choose the best colour. If we reverse it, it's just a discussion of certain people's imaginations of something,

**Comment by boothym** [comment]
@Adamant36 I think #844 would be good issue to have a look at, as there's already icons.

**Comment by boothym** [comment]
@Adamant36 also if you are doing test renderings, it would be great to see what railway=disused looks like on z14, which would be an easy fix to solve this issue: https://github.com/gravitystorm/openstreetmap-carto/issues/2074#issuecomment-354611437

**Comment by Adamant36** [comment]
@Tomasz-W, when we get a few issues completed can you start a new, similar issue that combines all the issues that are left in your various lists? As it is, this issue is getting messy and buried under other issues. Id like it to be more prominent and easier to find for the new coders that are coming along. Plus, its a becoming a hassle for me to sift through. Thanks.

**Comment by Adamant36** [comment]
@jeisenbe, I agree. I've been thinking about doing a PR for ridges for a while now but didn't want to do the other ones with it. So breaking it up would be good. The same goes for sport. Especially now that there is a PR for swimming. 

A few of the PRs like craft probably would have been done already if it didn't involve 20 different things in the same PR. I much rather see things get implemented at a slow drip pace then not at all. So, generally, the more we can split things up the better. Since it makes it a lot easier on everyone to keep track of, do the coding, test etc.

**Comment by Tomasz-W** [comment]
@Adamant36 I have a plan to open new ticket with task list from second post when all from the first one will be done. We can't close/ open some tickets just because they are lost in issues jungle, if we start bevaving like this, another users will start to close/ open tickets for the same reason and this mess would become even larger. If it is easier for coders to split PRs into smaller pieces, they should do it, but issues are something different and from my observation splitting issues into smaller pieces makes things worse (discussion usually becomes some unreadable web of references and mentions then)

Please add this link to your bookmarks, it should help you: https://github.com/gravitystorm/openstreetmap-carto/issues?q=is:issue+is:open+sort:updated-desc (issues sorted by last updates, it make things easier)

**Comment by Adamant36** [comment]
@Tomasz-W, however you want to do it. A few of the issues in the first list have stalled out though and I don't see them getting done any time soon if ever. Like the horse racing one. So hopefully if there is a few left that won't be done for a while it won't dissuade you from creating a new issue. 

As far as creating multiple issues for things goes, I agree with you in principle. It depends on the issue though. The sport issue already has off branching issues and multiple conversation strands/references (like rendering lines on pitches when the issue is about icons). So do other issues like #788 (which involves two clearly separate topics, how to render ridges and also if ocean names should be rendered as dots or not). Splitting them in the first place would have helped. For me, its more about encouraging people to split up new issues if possible going forward then doing it proactively. Also, we should encourage people more to create a new issue when the conversation goes off track but is still working on at some point (sports lines for instance). That way, good ideas that might not be addressed right then wont just get lost in the melee. Plus, its makes PRs a lot easier. A lot of times I don't give a rational for a PR unless I am asked for one just because I don't want to sift through a years worth of messages that are only vaguely related to each other, if at all, to figure out why the PR should be implemented. Or I'll just pass over issues that have to many parts. It might be the only one that does that, but either way, whatever we can do to make things easier for possible contributors the better I say. 


Anyway, thanks for the link. I didn't know there was a special page for that. I'll have to bookmark it instead of the main page.

**Comment by Tomasz-W** [comment]
@Adamant36 Speaking about issues from first post: they all need at least test renderings (to rate different icon designs or decide do we need any icon ), so "stalled out" discussion is propably caused by lack of them. I know it might be uncomfortable for you to work on something which can be never merged, but it's the price of fair and productive cooperation. There are many icons designed by me which was rejected, but I don't see working on them as a waste of time. Trying different things and testing them is a part of discussion, so if there was some users interested in adding some feature, there are no new comments, but there is also no test renderings, you shouldn't consider it as finished discussion.

I'm very satisfied with last few months of development, when you was working on listed issues one by one. If you would like to work on next ones splitting it into smaller PRs feel free to do it (just remember about adding "related to" words instead of "closes" before issue number in PR first post). I'll take care of lefting information for another users that only a part of some issue was resolved, so don't worry ;)

**Comment by Adamant36** [comment]
Ok. Its mostly a matter of time. Ive been really busy with school the last month. So im trying to pick things here to do when I have the time that will be productive. Plus, I'm just not much for all the debating thats happening lately. Its not really my strong suit. I much rather just be in the weeds coding and testing things. The good thing about a lot of the issues I worked on before was that they were mostly settled. So the discussion was minimal on them. It seems like we have mostly ran out of those types of issue though. 
I plan to do the sport icons when the swimming PR gets merged and I still need to do the testing etc on the other PRs I have going. I guess I dont mind things I work on not getting merged, but it is a time investment. Yeah, it can be disapointing if it happens, but I know thats life. Most of the time I learn way more from the things that fail on here then the ones that succeed. So, I guess its worth my time either way. 

Anyway, thanks for the pep talk. You make some good points. Plus, I appreciate that you'll update issues if I only resolve part of them. That would help a lot ;)

**Comment by Adamant36** [comment]
Btw, can you see whats going on with the horse icon issue and try to work it out so I can add it to the sports issue when I work on it? Both issues are pretty high on my priority list. I guess I could do the rest of the sports icons without it, but id like it to be in there if possible.

**Comment by lakedistrictOSM** [comment]
@jeisenbe @Adamant36 I've [suggested](https://github.com/gravitystorm/openstreetmap-carto/issues/3148#issuecomment-427436633) that #3148 is reopened so that it can be done separately from #788, if any of you would like to do it. :)

**Comment by Adamant36** [comment]
@Tomasz-W, id like to see a new clean issue that incorporates issues from my list and your further objectives list. So it might be worth starting a new one now since 99% of the issue in the first post are either done, PR's, or closed and not going anywhere. Plus, there is some done on your further objectives list also. 

It would save a lot of time with scrolling and sifting through closed/done/PRed issues.

**Comment by kocio-pl** [comment]
Maybe you need a wiki to not scroll and rewrite tickets all the time? GitHub has such a module for projects, you should ask @gravitystorm to switch it on then.

**Comment by Adamant36** [comment]
Sounds like a solid idea. You should do that @Tomasz-W.

**Comment by Tomasz-W** [comment]
@Adamant36 I promise I will open fresh one at the end of 2018. As I said before, I'm against starting 'close & open' repeated issues in random time behaviour, because I'm worried that then people will start do it in mass "because everybody forgot about my ticket", at the other hand I think that idea of opening new year with refreshed version of this overloaded meta-issue would be a good excuse for making an exeption ;)

**Comment by kocio-pl** [comment]
I have jusst changed the name of this ticket to better fit the place for pinned issues.

**Comment by Adamant36** [comment]
@kocio-pl, thank's for pinning it. It makes it a lot easier to find. 

@Tomasz-W, you want to strike out issue #1257 since its been closed?

**Comment by Tomasz-W** [comment]
@Adamant36 Thanks for noticing! I've updated the list.
 
PS. Small change - I'll open new version of this ticket in a first days of 2019 as I want to write a little summary of our work in 2018.

**Comment by kocio-pl** [comment]
No problem - when you open new version, I will just pin it instead of this one.

**Comment by Tomasz-W** [comment]
Big thanks for all involved in, especially coders! ❤️ As this issue began to be overloaded, I'm opening new one.

Please continue further discussion in https://github.com/gravitystorm/openstreetmap-carto/issues/3617.

**Comment by Tomasz-W** [comment]
Big thanks for all involved in, especially coders! ❤️ As this issue began to be overloaded, I'm opening new one.

Please continue further discussion in https://github.com/gravitystorm/openstreetmap-carto/issues/3617.
